filecoin-proofs/src/api/mod.rs

```diff
@@ -748,6 +756,9 @@ mod tests {
             pre_commit_output,
             &piece_infos,
         )?;
+
+        clear_cache(cache_dir.path())?;
```

Comment:
You seem to call `clear_cache` always after the `seal_commit_phase1()`. Would it make sense to just put it into the `seal_commit_phase1()` call?

Reply:
Nevermind. Given the commit message, it's only cleared for tests/benchmarks. That makes sense.

Reply:
But it's removed from `generate_candidates()`. I'm confused :)

Reply:
@vmx All good points.  @dignifiedquire mentioned that it should not be in generate candidates since that's called very frequently.  Really, we just need to be sure it's called sometime before then (not strictly, but ideally).  There is a clear_temp call in seal_commit_phase1, but there were some test paths that reached generate_candidates where that wasn't called for whatever reason, so I've been trying to track those down and make sure it's added everywhere explicitly before then.  If there's a saner way to do this, I'm open to it!